src/mpi/coll/ialltoall/ialltoall.c

```diff
@@ -288,6 +288,10 @@ int MPIR_Ialltoall_impl(const void *sendbuf, int sendcount,
                 mpi_errno =
                     MPIR_Ialltoall_intra_gentran_ring(sendbuf, sendcount, sendtype, recvbuf,
                                                       recvcount, recvtype, comm_ptr, request);
+                if (mpi_errno)
+                    MPIR_ERR_POP(mpi_errno);
+                goto fn_exit;
```

Comment:
`mpi_errno` don't need be checked as it is directly returned to the calling function and being checked there. If you agree, I'll simply remove that and sign-off and commit. (otherwise, reply quickly :) )

Reply:
Yeah, it is better idea to roll this int #3606. I'll do that instead.